packages/augur-sdk/src/api/Gnosis.ts

```diff
@@ -305,32 +301,24 @@ export class Gnosis {
         setupData,
       });
 
-      this.safesToCheck.push({
-        safe: ethUtil.toChecksumAddress(response.safe),
-        owner: params.owner,
-        status: GnosisSafeState.WAITING_FOR_FUNDS,
-      });
-
-      this.augur.setGnosisStatus(GnosisSafeState.WAITING_FOR_FUNDS);
-
-      return response;
-    } catch(error) {
-      const restoreAddress = await this.calculateGnosisSafeAddress({
-        payment: '1',
+      const calculatedSafeAddress = await this.calculateGnosisSafeAddress({
+        ...response,
         owner: params.owner,
-        safe: null,
-        paymentToken: null,
       });
+      if(ethUtil.toChecksumAddress(calculatedSafeAddress) !== ethUtil.toChecksumAddress(response.safe)) {
+        this.augur.setGnosisStatus(GnosisSafeState.ERROR);
+        throw new Error(`Potential malicious relay. Returned ${response.safe}. Expected ${calculatedSafeAddress}`);
+      }
 
       this.safesToCheck.push({
-        safe: ethUtil.toChecksumAddress(restoreAddress),
+        safe: ethUtil.toChecksumAddress(response.safe),
```

Comment:
To avoid someone accidentally introducing a bug later, I would recommend changing this to `calculatedSafeAddress`.  We can assume in the current code that this is the same, but should the above conditional ever change that assertion would no longer hold.  To avoid accidentally introducing a security vulnerability it feels prudent to just use the version that we are confident is good.